src/Access/DiskAccessStorage.cpp

```diff
@@ -386,21 +379,85 @@ bool DiskAccessStorage::rebuildLists()
         if (!entity)
             continue;
 
-        const String & name = entity->getName();
-        auto type = entity->getType();
-        auto & entry = entries_by_id[id];
-        entry.id = id;
-        entry.type = type;
-        entry.name = name;
-        entry.entity = entity;
-        auto & entries_by_name = entries_by_name_and_type[static_cast<size_t>(type)];
-        entries_by_name[entry.name] = &entry;
+        all_entities.emplace_back(id, entity);
     }
 
+    setAll(all_entities);
+
     for (auto type : collections::range(AccessEntityType::MAX))
         types_of_lists_to_write.insert(type);
 
-    return true;
+    failed_to_write_lists = false; /// Try again writing lists.
+    writeLists();
+}
+
+void DiskAccessStorage::setAll(const std::vector<std::pair<UUID, AccessEntityPtr>> & all_entities)
+{
+    /// This function is similar to MemoryAccessStorage::setAll().
+    boost::container::flat_set<UUID> not_used_ids;
+    std::vector<UUID> conflicting_ids;
+
+    /// Get the list of currently used IDs. Later we will remove those of them which are not used anymore.
+    for (const auto & id : entries_by_id | boost::adaptors::map_keys)
+        not_used_ids.emplace(id);
```

Comment:
what about
```suggestion
    for (const auto & [id, _] : entries_by_id)
        not_used_ids.emplace(id);
```

Reply:
As far as I know we can't skip a value in a structured binding with `_`, can we?

Reply:
Earlier this approach also showed warning that `_` is an unused variable, let's see what will happen this time